rocAL/rocAL/source/tf_meta_data_reader_detection.cpp

```diff
@@ -89,17 +88,16 @@ void TFMetaDataReaderDetection::lookup(const std::vector<std::string> &image_nam
             img_size.w = img_size.h =0;
             bb_coords.push_back(box);
             bb_labels.push_back(0);
-            img_sizes.push_back(img_size);
             // bb_coords={};
 
 	    _output->get_bb_cords_batch()[i] = bb_coords;
 	    _output->get_bb_labels_batch()[i] = bb_labels;
-        _output->get_img_sizes_batch()[i] = img_sizes;
+        _output->get_img_sizes_batch()[i] = img_size;
         }
 	else{
         _output->get_bb_cords_batch()[i] = it->second->get_bb_cords();
         _output->get_bb_labels_batch()[i] = it->second->get_bb_labels();
-        _output->get_img_sizes_batch()[i] = it->second->get_img_sizes();
+        _output->get_img_sizes_batch()[i] = it->second->get_img_size();
```

Comment:
do direct initialization here. Look at the case for SSD.

Reply:
@rrawther : is it changing the lookup function similar to lookup function of coco_meta_data_reader.cpp?
[https://github.com/sampath1117/MIVisionX/blob/rocAL_Keypoint_review/rocAL/rocAL/source/coco_meta_data_reader.cpp#L54](https://github.com/sampath1117/MIVisionX/blob/rocAL_Keypoint_review/rocAL/rocAL/source/coco_meta_data_reader.cpp#L54)

Reply:
Hi Sampath, I was referring to line #93 and #94 where it is initializing with label 0 and fixed bbox. There you can directly initialize the _output->get_bb_cords_batch()[i] with const values instead of using a temporary bb_coords struct

Reply:
Okay , Thanks Rajy for the clarification
I have made this change in latest commits to the PR